codec.go

```diff
@@ -125,7 +125,7 @@ func (u *UUID) decodeCanonical(t []byte) error {
 		return fmt.Errorf("uuid: incorrect UUID format %s", t)
 	}
 
-	src := t[:]
```

Comment:
Could this be an optimization? I recall a talk from today talking about "this being on the stack" now IIRC.

Can we benchmark before and after?

Reply:
Both are values on stack I presume

Reply:
The t value is already a slice i think. The `[:]` should be a no-op.

Reply:
It looks like a no-op indeed https://go.godbolt.org/z/OuMHTy

Reply:
What's the reason for even doing it? Might it make more sense just to put `src` in the function declaration and omit the line altogether? It's not a breaking API change.

Reply:
@theckman Unfortunately, that would mean we lose consistency with the rest of the `decodeX` methods in that file, which all take a `t []byte` parameter.

FWIW, I have no opinion on this. I think the proposed change is fine as-is.